### ISSUE TITLE
version 1.71.39 breaks on my ubuntu servers

### ISSUE THREAD

**Issue by nevs55** [tailscale/tailscale#12975]
### What is the issue?

upgrading to unstable release 1.71.39 breaks and will not start. I have to roll back to 1.71.17

### Steps to reproduce

after rolling back to 1.71.17 I tried to upgrade again and same issue

### Are there any recent changes that introduced the issue?

no

### OS

Linux

### OS version

Ubuntu 22.04

### Tailscale version

1.71.39

### Bug report

BUG-d18b4c3939ee0e30bc43cdc0ac1b06922e8768016d4f33e59de3ddd94d179b07-20240730205705Z-05d7b4174bea6aaf

**Comment by moderation** [comment]
Same on both OpenSUSE Tumbleweed on amd64 and Debian Bookwork on aarch64 - https://x.com/moderat10n/status/1818382542577541207

**Comment by JohnFawkes** [comment]
same. ubuntu 22.04.4 output of ``tailscale up``

```
failed to connect to local tailscaled (which appears to be running as tailscaled, pid 27043). Got error: Failed to connect to local Tailscale daemon for /localapi/v0/status; systemd tailscaled.service not running. Error: dial unix /var/run/tailscale/tailscaled.sock: connect: no such file or directory
```

**Comment by irbekrm** [comment]
I wasn't able to reproduce this on my Ubuntu 22.04.

Do you folks see the same issue with `v1.71.41`? https://pkgs.tailscale.com/unstable/?v=1.71.41

**Comment by mnordhoff** [comment]
Quick response, and I don't know the right way to debug this.

1.71.39 also crashes on startup for me on Ubuntu 20.04.

Haven't tried 1.71.41 yet.

I was on 1.71.17 before.

I got a bugreport **after downgrading again** in case there's anything useful got sent to the mothership. BUG-8a765d4bf3a9b7e3398c745fce93d3aa0729a7c712724315313045c45b8965ae-20240731090820Z-d24dedd61d1a6a5e

Logs start with the below; it continued but looked repetitive. Seems `net/dns/resolver` is dying with `panic: nil health`. Might be notable that I have all the DNS stuff off.

```
Jul 31 09:05:40 ultimate systemd[1]: Stopping Tailscale node agent...
Jul 31 09:05:40 ultimate tailscaled[1462267]: tailscaled got signal terminated; shutting down
Jul 31 09:05:40 ultimate tailscaled[1462267]: control: client.Shutdown ...
Jul 31 09:05:40 ultimate tailscaled[1462267]: control: mapRoutine: exiting
Jul 31 09:05:40 ultimate tailscaled[1462267]: control: authRoutine: exiting
Jul 31 09:05:40 ultimate tailscaled[1462267]: control: updateRoutine: exiting
Jul 31 09:05:40 ultimate tailscaled[1462267]: control: Client.Shutdown done.
Jul 31 09:05:40 ultimate tailscaled[1462267]: magicsock: closing connection to derp-16 (conn-close), age 11h46m48s
Jul 31 09:05:40 ultimate tailscaled[1462267]: magicsock: 0 active derp conns
Jul 31 09:05:40 ultimate tailscaled[1462267]: monitor: RTM_DELROUTE: src=, dst=fe80::/64, gw=, outif=53, table=254
Jul 31 09:05:40 ultimate tailscaled[1462267]: monitor: RTM_DELROUTE: src=, dst=fe80::4735:a890:48f:7ccb/128, gw=, outif=53, table=255
Jul 31 09:05:40 ultimate tailscaled[1462267]: monitor: RTM_DELROUTE: src=, dst=ff00::/8, gw=, outif=53, table=255
Jul 31 09:05:40 ultimate tailscaled[1462267]: monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:254 Protocol:0 Scope:0 Type:1 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5210 Table:254 Mark:16711680 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
Jul 31 09:05:40 ultimate tailscaled[1462267]: monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:253 Protocol:0 Scope:0 Type:1 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5230 Table:253 Mark:16711680 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
Jul 31 09:05:40 ultimate tailscaled[1462267]: monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:0 Protocol:0 Scope:0 Type:7 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5250 Table:0 Mark:16711680 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
Jul 31 09:05:40 ultimate tailscaled[1462267]: monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:52 Protocol:0 Scope:0 Type:1 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5270 Table:52 Mark:0 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
Jul 31 09:05:40 ultimate tailscaled[1462267]: flushing log.
Jul 31 09:05:40 ultimate tailscaled[1462267]: logger closing down
Jul 31 09:05:40 ultimate tailscaled[903154]: logtail started
Jul 31 09:05:40 ultimate tailscaled[903154]: Program starting: v1.71.39-teead25560-ga4fd279e8, Go 1.22.5: []string{"/usr/sbin/tailscaled", "--cleanup"}
Jul 31 09:05:40 ultimate tailscaled[903154]: LogID: 8a765d4bf3a9b7e3398c745fce93d3aa0729a7c712724315313045c45b8965ae
Jul 31 09:05:40 ultimate tailscaled[903154]: logpolicy: using $STATE_DIRECTORY, "/var/lib/tailscale"
Jul 31 09:05:40 ultimate tailscaled[903154]: dns: [rc=unknown ret=direct]
Jul 31 09:05:40 ultimate tailscaled[903154]: dns: using "direct" mode
Jul 31 09:05:40 ultimate tailscaled[903154]: flushing log.
Jul 31 09:05:40 ultimate tailscaled[903154]: logger closing down
Jul 31 09:05:40 ultimate systemd[1]: tailscaled.service: Control process exited, code=exited, status=2/INVALIDARGUMENT
Jul 31 09:05:40 ultimate systemd[1]: tailscaled.service: Failed with result 'exit-code'.
Jul 31 09:05:40 ultimate systemd[1]: Stopped Tailscale node agent.
Jul 31 09:05:40 ultimate systemd[1]: Starting Tailscale node agent...
Jul 31 09:05:40 ultimate tailscaled[903164]: logtail started
Jul 31 09:05:40 ultimate tailscaled[903164]: Program starting: v1.71.39-teead25560-ga4fd279e8, Go 1.22.5: []string{"/usr/sbin/tailscaled", "--state=/var/lib/tailscale/tailscaled.state", "--socket=/run/tailscale/tailscaled.sock", "--port=41641"}
Jul 31 09:05:40 ultimate tailscaled[903164]: LogID: 8a765d4bf3a9b7e3398c745fce93d3aa0729a7c712724315313045c45b8965ae
Jul 31 09:05:40 ultimate tailscaled[903164]: logpolicy: using $STATE_DIRECTORY, "/var/lib/tailscale"
Jul 31 09:05:40 ultimate tailscaled[903164]: RAW-STDERR: ***
Jul 31 09:05:40 ultimate tailscaled[903164]: RAW-STDERR: *** Lines prefixed with RAW-STDERR below bypassed logtail and probably come from a previous run of the program
Jul 31 09:05:40 ultimate tailscaled[903164]: RAW-STDERR: ***
Jul 31 09:05:40 ultimate tailscaled[903164]: RAW-STDERR:
Jul 31 09:05:40 ultimate tailscaled[903164]: RAW-STDERR: [{"logtail":{"client_time":"2024-07-31T09:05:40.414725713Z","proc_id":2605857443,"proc_seq":5},"text":"dns: [rc=unknown ret=direct]\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,{"logtail":{"client_time":"2024-07-31T09:05:40.415171059Z","proc_id":2605857443,"proc_seq":6},"text":"dns: using \"direct\" mode\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,{"logtail":{"client_time":"2024-07-31T09:05:40.415626465Z","proc_id":2605857443,"proc_seq":7},"text":"flushing log.\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,{"logtail":{"client_time":"2024-07-31T09:05:40.415883099Z","proc_id":2605857443,"proc_seq":8},"text":"logger closing down\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,RAW-STDERR: [{"logtail":{"client_time":"2024-07-31T09:05:40.414725713Z","proc_id":2605857443,"proc_seq":5},"text":"dns: [rc=unknown ret=direct]\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,{"logtail":{"client_time":"2024-07-31T09:05:40.415171059Z","proc_id":2605857443,"proc_seq":6},"text":"dns: using \"direct\" mode\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,{"logtail":{"client_time":"2024-07-31T09:05:40.415626465Z","proc_id":2605857443,"proc_seq":7},"text":"flushing log.\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,{"logtail":{"client_time":"2024-07-31T09:05:40.415883099Z","proc_id":2605857443,"proc_seq":8},"text":"logger closing down\n"}
Jul 31 09:05:40 ultimate tailscaled[903164]: ,{"text":"panic: nil health\n"}
```

**Comment by irbekrm** [comment]
>  Seems net/dns/resolver is dying with panic: nil health. Might be notable that I have all the DNS stuff off

@mnordhoff I think this got fixed in https://github.com/tailscale/tailscale/pull/12969

**Comment by irbekrm** [comment]
.. and I think `v1.71.41` will have the fix - let us know if it still breaks

**Comment by JohnFawkes** [comment]
> I wasn't able to reproduce this on my Ubuntu 22.04.
> 
> Do you folks see the same issue with `v1.71.41`? https://pkgs.tailscale.com/unstable/?v=1.71.41
> 
> If so, could you clarify what version you upgraded from and from what state (running or stopped tailscale)?

Havent tried .41 yet but I upgraded from 1.71.17 to 1.71.39 and it broke as soon as it installed. I havent changed any settings or anything pretty much since Ive had tailscale installed on my home server in like 6 months or more.

**Comment by irbekrm** [comment]
@JohnFawkes I'm sorry, this is my bad, we actually did not build that version for the linux client 🤦🏼 I will trigger the builds post the version that has the fix once we have it

**Comment by irbekrm** [comment]
> @JohnFawkes I'm sorry, this is my bad, we actually did not build that version for the linux client 🤦🏼 I will trigger the builds and post the version that has the fix once we have it

Linux client packages should now be available for the `1.71.41` version. Please let us know if this fixes the issue

**Comment by mnordhoff** [comment]
1.71.41 works for me. :+1:

**Comment by nevs55** [comment]
Yes 1.71.41 has fixed the issue

**Comment by irbekrm** [comment]
I'm going to close this as it seems that the bug was fixed in `1.71.41`. Thanks a lot for opening the issue and providing the feedback and extra info folks 🙏🏼
